packages/react/src/Panels/WorkspacePanel.tsx

```diff
@@ -32,6 +32,13 @@ interface TerminalProps extends PanelProps {
  * This component is the orchestrator between various interactive components.
  */
 export function WorkspacePanel({ tutorialStore, theme }: Props) {
+  /**
+   * Re-render when lesson changes.
+   * The `tutorialStore.hasEditor()` and other methods below access
+   * stale data as they are not reactive.
+   */
+  useStore(tutorialStore.ref);
+
   const hasEditor = tutorialStore.hasEditor();
   const hasPreviews = tutorialStore.hasPreviews();
   const hideTerminalPanel = !tutorialStore.hasTerminalPanel();
```

Comment:
Ideally we should make these reactive instead, e.g.

```ts
const hasEditor = useStore(tutorialStore.hasEditor);
```

... but as that would require quite much refactoring on the store, e.g. making `lesson` reactive, I'd leave that out side of this PR.